### PR TITLE
wl_display_roundtrip() must be called on the Wayland thread

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fixes: #1775

### CODE REVIEW

File: src/platforms/wayland/display.cpp [hunk 1]
```diff
@@ -269,15 +276,28 @@ try
...
     }
 }
 catch (std::exception const& e)
 {
     fatal_error("Critical error in Wayland platform: %s\n", boost::diagnostic_information(e).c_str());
 }
 
+void mir::graphics::wayland::Display::roundtrip() const
+{
+    eventfd_write(roundtrip_signal, 1);
```
Comment (wmww) [inline src/platforms/wayland/display.cpp:295]: Does this block until the roundtrip has been completed? How? If not, what's the point of calling it at all?

Comment (AlanGriffiths) [reply src/platforms/wayland/display.cpp:295]: No. That happens on the Wayland thread.

The point is that this empties the queue of message to the server. (Which needs to happen on the Wayland thread.) Is there a better way?

Comment (AlanGriffiths) [reply src/platforms/wayland/display.cpp:295]: /me reads the documentation and finds "wl_display_flush()"

File: src/platforms/wayland/cursor.cpp [hunk 2]
```diff
@@ -66,10 +66,9 @@ struct wl_shm_pool* make_shm_pool(struct wl_shm* shm, int size, void **data)
 }
 }
 
-mpw::Cursor::Cursor(wl_display* display, wl_compositor* compositor, wl_shm* shm, std::function<void()> roundtrip) :
+mpw::Cursor::Cursor(wl_display* display, wl_compositor* compositor, wl_shm* shm, std::function<void()> flush) :
     display{display},
-    shm{shm},
-    roundtrip{std::move(roundtrip)},
+    shm{shm}, flush{std::move(flush)},
```
Comment (wmww) [inline src/platforms/wayland/cursor.cpp]: Formatting

File: src/platforms/wayland/display.cpp [hunk 3]
```diff
@@ -88,7 +88,7 @@ mgw::Display::Display(
     DisplayClient{wl_display, gl_config},
     report{report},
     shutdown_signal{::eventfd(0, EFD_CLOEXEC)},
-    roundtrip_signal{::eventfd(0, EFD_SEMAPHORE)},
+      flush_signal{::eventfd(0, EFD_SEMAPHORE)},
```
Comment (wmww) [inline src/platforms/wayland/display.cpp:91]: Formatting

**Review by wmww — APPROVED** [review]
Just some formatting nits
bors delegate+

**Comment by AlanGriffiths** [comment]
bors r=wmww

**Comment by AlanGriffiths** [comment]
bors r=wmww

**Pull request merged**